crates/bevy_ecs/src/world/archetype_invariants.rs

```diff
@@ -453,24 +455,18 @@ impl ArchetypeInvariants {
                     }
                 }
 
-                let archetype_component_names: Vec<String> = component_ids_of_archetype
-                    .into_iter()
-                    .map(|id| match components.get_info(id) {
-                        Some(info) => get_short_name(info.name()),
-                        None => format!("{:?}", id),
-                    })
-                    .collect();
+                let archetype_component_names =
+                    display_component_id_types(component_ids_of_archetype.iter(), components);
 
-                let failed_invariant_names: String = failed_invariants
-                    .into_iter()
+                let failed_invariant_names = failed_invariants
+                    .iter()
                     .map(|invariant| invariant.display(components))
                     .reduce(|acc, s| format!("{}\n{}", acc, s))
-                    .unwrap();
+                    .unwrap_or_default();
 
                 panic!(
-                    "Archetype invariant violated! The following invariants were violated for archetype {:?}:\n{}",
-                    archetype_component_names,
-                    failed_invariant_names,
+                    "At least one archetype invariant was violated for the archetype [{archetype_component_names}]: \
+                    \n{failed_invariant_names}"
```

Comment:
The panic message is not consistent with `UntypedArchetypeInvariant`.

Reply:
This is somewhat intentional- in the method on `UntypedArchetypeInvariant`, we're only testing one invariant (and so only one invariant can fail). In this error message, there may be multiple failing invariants to output. That being said, if you have any ideas to improve the wording consistency, I'd love to hear them :)